src/html_builder.ts

```diff
@@ -0,0 +1,503 @@
+        const inner_lft: number = outer_rad - inner_rad;
+
+        const cl_opacity: number = 0.4;
+        let tbid: string = ulabel.config.toolbox_id;
+
+        const center_coord: number = width / 2;
+
+        for (const st in ulabel.subtasks) {
+            const idd_id = ulabel.subtasks[st]["state"]["idd_id"];
+            const idd_id_front = ulabel.subtasks[st]["state"]["idd_id_front"];
+
+            let subtask_dialog_container_jq = $("#dialogs__" + st);
+            let front_subtask_dialog_container_jq = $("#front_dialogs__" + st);
+
+            let dialog_html_v2 = HTMLBuilder.get_idd_string(
+                idd_id, width, center_coord, cl_opacity, ulabel.subtasks[st]["class_ids"],
+                inner_rad, outer_rad, ulabel.subtasks[st]["class_defs"]
+            );
+            let front_dialog_html_v2 = HTMLBuilder.get_idd_string(
+                idd_id_front, width, center_coord, cl_opacity, ulabel.subtasks[st]["class_ids"],
+                inner_rad, outer_rad, ulabel.subtasks[st]["class_defs"]
+            );
+
+            // TODO noconflict
+            let toolbox_html: string = `<div id="tb-id-app--${st}" class="tb-id-app">`;
+            const class_ids = ulabel.subtasks[st]["class_ids"];
+
+
+            for (let i = 0; i < class_ids.length; i++) {
+
+                let this_id: string = class_ids[i];
+                let this_color: string = ulabel.subtasks[st]["class_defs"][i]["color"];
+                let this_name: string = ulabel.subtasks[st]["class_defs"][i]["name"];
+
+                let sel: string = "";
+                let href: string = ' href="#"';
+                if (i == 0) {
+                    sel = " sel";
+                    href = "";
+                }
+                if (ulabel.config["allow_soft_id"]) {
+                    let msg = "Only hard id is currently supported";
+                    throw new Error(msg);
+                }
+                else {
+                    toolbox_html += `
+                        <a${href} id="${tbid}_sel_${this_id}" class="tbid-opt${sel}">
+                            <div class="colprev ${tbid}_colprev_${this_id}" style="background-color: ${this_color}"></div> <span class="tb-cls-nam">${this_name}</span>
+                        </a>
+                    `;
+                }
+            }
+            toolbox_html += `
+            </div>`;
+
+            // Add dialog to the document
+            // front_subtask_dialog_container_jq.append(dialog_html);
+            // $("#" + ul.subtasks[st]["idd_id"]).attr("id", ul.subtasks[st]["idd_id_front"]);
+            front_subtask_dialog_container_jq.append(front_dialog_html_v2); // TODO(new3d) MOVE THIS TO GLOB BOX -- superimpose atop thee anchor already there when needed, no remove and add back
+            subtask_dialog_container_jq.append(dialog_html_v2);
+            // console.log(dialog_html);
+            // console.log(dialog_html_v2);
```

Comment:
clean up comments

Reply:
Resolved in [c3aeb0d](https://github.com/SenteraLLC/ulabel/pull/101/commits/c3aeb0d95acf2ec0bfeb90a02b71b558a25bc8bd)